src/Shouldly/Internals/EqualityComparer.cs

```diff
@@ -26,20 +25,11 @@ public EqualityComparer(IEqualityComparer? innerComparer = null)
 
         public bool Equals([AllowNull] T x, [AllowNull] T y)
         {
-            var type = typeof(T);
-
             if (ReferenceEquals(x, y))
                 return true;
 
-            // Null?
-            if (!type.IsValueType() || (type.IsGenericType() && type.GetGenericTypeDefinition().IsAssignableFrom(NullableType)))
-            {
-                if (Equals(x, null))
-                    return Equals(y, null);
-
-                if (Equals(y, null))
-                    return false;
```

Comment:
Object.Equals returns false if *either* parameter is null. Therefore these are equivalent to `x is null` and `y is null`.

`y is null` is false if `x is null` because of the ReferenceEquals check at the beginning.

The `if (!type.IsValueType() || (type.IsGenericType() && type.GetGenericTypeDefinition().IsAssignableFrom(NullableType)))` check is a very long way to check `default(T) is null`. Neither check is needed because if `default(T)` is not null, then `x` and `y` cannot be null and therefore the null checks are no-ops. (They are even elided during jit.)